exercises/proverb/canonical-data.json

```diff
@@ -2,97 +2,93 @@
-                        "For want of a soldier the battle was lost.",
-                        "And all for the want of a pin."
-                    ],
-                    "input": [
-                        "pin",
-                        "gun",
-                        "soldier",
-                        "battle"
-                    ],
-                    "property": "recite"
-                }
+            "description": "zero pieces",
+            "property": "recite",
+            "input": [],
+            "expected": []
+        },
+        {
+            "description": "one piece",
+            "property": "recite",
+            "input": [
+                "nail"
+            ],
+            "expected": [
+                "And all for the want of a nail."
+            ]
+        },
+        {
+            "description": "two pieces",
+            "property": "recite",
+            "input": [
+                "nail",
+                "shoe"
+            ],
+            "expected": [
+                "For want of a nail the shoe was lost.",
+                "And all for the want of a nail."
+            ]
+        },
+        {
+            "description": "three pieces",
+            "property": "recite",
+            "input": [
+                "nail",
+                "shoe",
+                "horse"
+            ],
+            "expected": [
+                "For want of a nail the shoe was lost.",
+                "For want of a shoe the horse was lost.",
+                "And all for the want of a nail."
+            ]
+        },
+        {
+            "description": "full proverb",
+            "property": "recite",
+            "input": [
+                "nail",
+                "shoe",
+                "horse",
+                "rider",
+                "message",
+                "battle",
+                "kingdom"
+            ],
+            "expected": [
+                "For want of a nail the shoe was lost.",
+                "For want of a shoe the horse was lost.",
+                "For want of a horse the rider was lost.",
+                "For want of a rider the message was lost.",
+                "For want of a message the battle was lost.",
+                "For want of a battle the kingdom was lost.",
+                "And all for the want of a nail."
+            ]
+        },
+        {
+            "description": "four pieces modernized",
+            "property": "recite",
+            "input": [
+                "pin",
+                "gun",
+                "soldier",
+                "battle"
             ],
-            "description": "Construct a proverb of the form 'for want of an X, the Y was lost'"
+            "expected": [
+                "For want of a pin the gun was lost.",
+                "For want of a gun the soldier was lost.",
+                "For want of a soldier the battle was lost.",
+                "And all for the want of a pin."
```

Comment:
Is it to be presumed that the proverb is not set in stone?  The description only refers to a single proverb and makes not mention that the `"input"` provided might refer to a different proverb.  I don't think `"four pieces modernized"` should be included unless the description is also updated.

Reply:
@coriolinus my apologies!  I didn't read closely enough.  You are passing the expected proverb.

Reply:
@rpottsoh I did update the description [for that](https://github.com/exercism/problem-specifications/pull/994/files#diff-a82423102a1295ada2179ae4ce091d03R15): "your solution should be able to handle lists of arbitrary length and content". 

In short, the idea is to ensure that students won't just hard-code the entire poem as a list, and then return its last N items, where N is the length of the input list. The goal of the exercise is to ensure that they can manipulate strings.

This also helps distinguish this exercise from others; discussion [here](https://github.com/exercism/problem-specifications/pull/994#discussion_r148731929) was in favor of keeping the input list.

Reply:
Thanks @coriolinus for clearing this up for me.  I have not worked this exercise yet so some of the what was getting past me.  Thanks for taking the time to produce the canonical data for this exercise.